mocksipipeline/instrument/optics/response.py

```diff
@@ -319,3 +323,71 @@ def get_wcs(self, observer, roll_angle=90 * u.deg):
             pc_matrix=pc_matrix,
             observer=observer,
         )
+
+    def psf(self, wavelength=None, mask_oversample=4, fft_oversample = 4):
+        mask = self.aperture.mask(oversample=mask_oversample)
+        f = self.design.focal_length
+
+        x = mask.coords['x'].values * u.micron
+        y = mask.coords['y'].values * u.micron
+        x_step = x[0, 1] - x[0, 0]
+
+        f_number = f / (x[0, -1] - x[0, 0])
+        f_number = f_number.to('')
+
+        if wavelength is None:
+            wavelength = self.wavelength
+
+        wfe = mask.data * 1e9 * 1j  # only the fresnel term depends on wavelength
+
+        if wavelength.size == 1:
+            wavelength = [wavelength, ]
+
+        psfs = []
+        for wv in wavelength:
+            dx = wv * f_number / fft_oversample
+            dx = dx.to(u.micron)
+
+            fresnel_wfe = 1 / (2 * wv * f) * (x ** 2 + y ** 2) + 0.j
+            fresnel_wfe = fresnel_wfe.value
+
+            wfe_total = wfe + fresnel_wfe
+
+            fresnel_psf = _wfe2psf(wfe_total, oversample=fft_oversample)
+            x_psf = x / x_step * dx
+            y_psf = y / x_step * dx
+
+            psfs.append(
+                xr.DataArray(
```

Comment:
Is there a particular reason not to just make one big `DataArray` at the end that also has a wavelength axis?